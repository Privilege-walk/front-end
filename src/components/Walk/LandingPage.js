import React from "react";
import Typography from '@mui/material/Typography';
import Grid from '@mui/material/Grid';
import Button from '@mui/material/Button';
import Container from '@mui/material/Container';
import { event } from "../../api/mockData/event";

export default function LandingPage({goNextPage}){
    
    return (
        <Container sx={{px : 6, mt: 5}}>
            <Grid container justifyContent='center' alignItems='center' direction='column' spacing={2}>
                <Grid item>
                    <Typography variant="h5" component="div">
                        {event.name}
                    </Typography>
                </Grid>

                <Grid item>
                    <Typography variant="p" component="div">
                        {event.description}
                    </Typography>
                </Grid>

                <Grid item sx={{ mt: 2 }}>
                    <Button onClick={goNextPage} variant="outlined">Start</Button>
                </Grid>
            </Grid>
        </Container>
    )
}