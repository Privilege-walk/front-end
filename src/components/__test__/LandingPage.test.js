import React from "react";
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import LandingPage from '../Walk/LandingPage';

test('renders without crashing',  () => {
    let utils;
    act (() => {
        utils = render( <LandingPage eventName="eventName" />);
    }); 
    const { asFragment } = utils;
    expect(asFragment()).toMatchSnapshot();
})