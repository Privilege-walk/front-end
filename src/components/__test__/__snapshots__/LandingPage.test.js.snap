// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-tqw8gr-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-ma8ek8-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          eventName
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-p css-37jdci-MuiTypography-root"
        >
          Waiting for the host to start the event. Your answers are anonymized and will not be stored by the host.
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-r4jj1p-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Waiting for host ...
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
